confmap/provider.go

```diff
@@ -101,17 +102,39 @@ func WithRetrievedClose(closeFunc CloseFunc) RetrievedOption {
 }
 
 // NewRetrieved returns a new Retrieved instance that contains the data from the raw deserialized config.
-func NewRetrieved(rawConf map[string]interface{}, opts ...RetrievedOption) (Retrieved, error) {
+// The rawConf can be one of the following types:
```

Comment:
Some of the fields we have on the Collector configuration don't have these types, e.g. `zapcore.Level` for the log level, or `configcompression.CompressionType` for the compression type. It's fine not to support them on the first version, but do we have a plan to support them eventually?

Reply:
`zapcore.Level` is the "unmarshaled" type, the config should be "string" correct? The provider should not "unmarshal" into objects, only into primitives. The next part is other component job.

Reply:
Right, that makes sense, thanks for clarifying